mobile/native/src/ln_dlc/mod.rs

```diff
@@ -121,16 +121,18 @@ pub const FUNDING_TX_WEIGHT_ESTIMATE: u64 = 220;
 ///
 /// We do not wait for the triggered task to finish, because the effect will be reflected
 /// asynchronously on the UI.
-pub async fn refresh_wallet_info() -> Result<()> {
+pub async fn refresh_wallet_info(full: bool) -> Result<()> {
     let node = crate::state::get_node();
     let wallet = node.inner.wallet();
 
     // Spawn into the blocking thread pool of the dedicated backend runtime to avoid blocking the UI
     // thread.
     let runtime = get_or_create_tokio_runtime()?;
     runtime.spawn_blocking(move || {
-        if let Err(e) = wallet.sync() {
-            tracing::error!("Manually triggered on-chain sync failed: {e:#}");
+        if full {
+            if let Err(e) = wallet.sync() {
+                tracing::error!("Manually triggered on-chain sync failed: {e:#}");
+            }
         }
```

Comment:
🔧 Instead of introducing the flag it's nicer to simply:

```rust
sync_lightning_wallet() // fast

keep_wallet_balance_and_history_up_to_date() // fast and will update the screen asap

sync() // slow

keep_wallet_balance_and_history_up_to_date()
```

Obviously the double `keep_wallet_balance_and_history_up_to_date` is a bit silly, but the overhead is minimal and it happens in the background anyway.

Eventually we should somehow split up `keep_wallet_balance_and_history_up_to_date` anyway.

Reply:
I added a dedicated `refresh_lightning_wallet` api and got rid of the full flag. Note, `refresh_lightning_wallet` is now blocking allowing us to await the lightning wallet to be refreshed.